packages/lesswrong/lib/collections/surveys/schema.ts

```diff
@@ -0,0 +1,43 @@
+import { getWithLoader } from "@/lib/loaders";
+import { accessFilterMultiple, resolverOnlyField } from "@/lib/utils/schemaUtils";
+
+const schema: SchemaType<"Surveys"> = {
+  name: {
+    type: String,
+    optional: false,
+    nullable: false,
+    canRead: ["guests"],
+    canCreate: ["admins"],
+    canUpdate: ["admins"],
+  },
+  questions: resolverOnlyField({
+    type: Array,
+    graphQLtype: "[SurveyQuestion!]!",
+    canRead: ["guests"],
+    resolver: async (survey: DbSurvey, _args: void, context: ResolverContext) => {
+      const {currentUser, SurveyQuestions} = context;
+      const questions = await getWithLoader(
+        context,
+        SurveyQuestions,
+        "surveyQuestionsBySurvey",
+        {surveyId: survey._id},
+        "surveyId",
+        survey._id,
+      );
+      const ordered = questions.sort((a, b) => a.order - b.order);
+      return accessFilterMultiple(currentUser, SurveyQuestions, ordered, context);
+    },
+    sqlResolver: ({field}) => `(
+      SELECT ARRAY_AGG(ROW_TO_JSON(sq.*))
+      FROM "SurveyQuestions" sq
+      WHERE sq."surveyId" = ${field("_id")}
+      GROUP BY sq."order"
+      ORDER BY sq."order" DESC
+    )`,
```

Comment:
Btw this is the subquery pattern I was talking about in dev check in a while ago, where if you were writing this query manually you would likely put the `GROUP BY` outside the top level query, but sql resolvers force you to do it as a subquery.

See e.g. [here](https://www.cybertec-postgresql.com/en/subqueries-and-performance-in-postgresql/) for a discussion of the potential performance problems caused by this. My understanding of this:
- In the worst case you can end up executing 1 extra query per row in a nested loop, and this is very bad for performance
- Some combination of caching, postgres being able to optimise this away, and small amounts of data mean it often isn't really a problem. But still it would be better to do it as a join outside if possible